OSDOCS-11340 OCP 4.14.33 Release Notes (#78903)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-14-33_{context}"]
=== RHSA-2024:4479 - {product-title} 4.14.33 bug fix and security update

Issued: 2024-07-17

{product-title} release 4.14.33, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:4479[RHSA-2024:4479] advisory. There are no RPM packages in this update.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.14.33 --pullspecs
----

[id="ocp-4-14-33-bug-fixes_{context}"]
==== Bug fixes

> * Previously, nodes that were booted using 4.1 and 4.2 bootimages for {product-title} got stuck during provisioning because the `machine-config-daemon-firstboot.service` had incompatible machine-config-daemon binary code. With this release, binary has been updated and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-36776[*OCPBUGS-36776*])

Reviewer: ```suggestion
* Previously, nodes that were started by using 4.1 and 4.2 boot images for {product-title} got stuck during provisioning because the `machine-config-daemon-firstboot.service` had incompatible machine-config-daemon binary code. With this release, the binary is updated and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-36776[*OCPBUGS-36776*])
```
- I replaced "booted" with "started" per IBM Style word usage
- I updated the tense to simple present tense per IBM Style.

> * Previously, a change of dependency targets was introduced in {product-title} 4.14 that prevented disconnected ARO installs from scaling up new nodes after they upgraded to affected versions. With this release, disconnected ARO installs can scale up new nodes after upgrading to {product-title} 4.14. (link:https://issues.redhat.com/browse/OCPBUGS-36593[*OCPBUGS-36593*])

Reviewer: ```suggestion
* Previously, a change of dependency targets was introduced in {product-title} 4.14 that prevented disconnected ARO installations from scaling up new nodes after they upgraded to affected versions. With this release, disconnected ARO installs can scale up new nodes after upgrading to {product-title} 4.14. (link:https://issues.redhat.com/browse/OCPBUGS-36593[*OCPBUGS-36593*])
```

> * Previously, if a new deployment was performed at the OSTree level on a host that was identical to the current deployment but on a different stateroot, the OSTree saw them as equal. This behavior incorrectly prevented the boot loader from updating when `set-default` was invoked, as OSTree did not recognize the two stateroots as a differentiation factor for deployments. With this release, OSTree's logic has been modified to consider the stateroots and allows OSTree to properly set the default deployment to a new deployment with different stateroots. (link:https://issues.redhat.com/browse/OCPBUGS-36437[*OCPBUGS-36437*])

Reviewer: ```suggestion
* Previously, if a new deployment was completed at the OSTree level on a host that was identical to the current deployment but on a different stateroot, the OSTree saw them as equal. This behavior incorrectly prevented the boot loader from updating when `set-default` was invoked, as OSTree did not identify the two stateroots as a differentiation factor for deployments. With this release, the OSTree logic is modified to analyze the stateroots and OSTree can properly set the default deployment to a new deployment with different stateroots. (link:https://issues.redhat.com/browse/OCPBUGS-36437[*OCPBUGS-36437*])
```
- Removed anthropomorphic language: https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism
- Swapped out "performed" for "completed, per IBM Style word usage

* Previously, the `HighOverallControlPlaneCPU` alert triggered warnings based on criteria for multi-node clusters with high availability. As a result, misleading alerts were triggered in {sno} clusters because the configuration did not match the environment criteria. This update refines the alert logic to use {sno}-specific queries and thresholds and account for workload partitioning settings. As a result, CPU utilization alerts in {sno} clusters are accurate and relevant to single-node configurations. (link:https://issues.redhat.com/browse/OCPBUGS-31354[*OCPBUGS-31354*])

> * Previously, the DNS Operator did not verify that a cluster had ready nodes with available CPU in at least two availability zones, and the DNS daemon set did not use surge for rolling updates. As a result, clusters in which all nodes were in the same availability zone would repeatedly emit `TopologyAwareHintsDisabled` events for the cluster DNS service. With this release, the `TopologyAwareHintsDisabled` events are no longer emitted on clusters that do not have nodes in multiple availability zones and the issue has been resolved. (link:https://issues.redhat.com/browse/OCPBUGS-5943[*OCPBUGS-5943*])

Reviewer: ```suggestion
* Previously, the DNS Operator did not verify that a cluster had ready nodes with available CPU in at least two availability zones, and the DNS daemon set did not use surge for rolling updates. As a result, clusters in which all nodes were in the same availability zone repeatedly emitted `TopologyAwareHintsDisabled` events for the cluster DNS service. With this release, the `TopologyAwareHintsDisabled` events are no longer emitted on clusters that do not have nodes in multiple availability zones, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-5943[*OCPBUGS-5943*])
```

[id="ocp-4-14-33-updating_{context}"]
==== Updating

> To update an existing {product-title} 4.14 cluster to this latest release, see xref:../updating/updating_a_cluster/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI].

Reviewer: ```suggestion
To update an {product-title} 4.14 cluster to this latest release, see xref:../updating/updating_a_cluster/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI].
```

Author: Good catch!

Author: This language has been in the release notes for a long time so I'm reluctant to change it.
